Crash When Creating Cosmosdb With MongoDB Api In East US 2 Region

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

0.11.3

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_cosmosdb_account`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
https://drive.google.com/file/d/1poVwGmBjl3YIVVmzuOg1hyTm75THhLdh/view?usp=sharing
```

### Debug Output

[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 2019/05/24 23:04:10 [INFO] preparing arguments for AzureRM Cosmos DB Account creation.
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 2019/05/24 23:04:10 [INFO] preparing arguments for AzureRM Log Analytics Workspace creation.
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 2019/05/24 23:04:10 [DEBUG] AzureRM Request:
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: HEAD /providers/Microsoft.DocumentDB/databaseAccountNames/dev-eus2-store-cdb?api-version=2015-04-08 HTTP/1.1
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Host: management.azure.com
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: User-Agent: Go/go1.11.6 (amd64-linux) go-autorest/v11.7.0 Azure-SDK-For-Go/v29.0.0 documentdb/2015-04-08 Terraform/0.12.0 terraform-provider-azurerm/1.28.0
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 2019/05/24 23:04:10 [DEBUG] AzureRM Request:
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: PUT /subscriptions/2f8b822f-17f6-463f-b84a-f59bee254fe9/resourcegroups/dev-eus2-store-rg/providers/Microsoft.OperationalInsights/workspaces/dev-eus2-store-la?api-version=2015-11-01-preview HTTP/1.1
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Host: management.azure.com
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: User-Agent: Go/go1.11.6 (amd64-linux) go-autorest/v11.7.0 Azure-SDK-For-Go/v29.0.0 operationalinsights/2015-11-01-preview Terraform/0.12.0 terraform-provider-azurerm/1.28.0
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Content-Length: 183
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Content-Type: application/json; charset=utf-8
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Accept-Encoding: gzip
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 2019/05/24 23:04:11 [DEBUG] AzureRM Response for https://management.azure.com/providers/Microsoft.DocumentDB/databaseAccountNames/dev-eus2-store-cdb?api-version=2015-04-08:
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: HTTP/2.0 404 Not Found
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Content-Length: 97
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Cache-Control: no-store, no-cache
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Date: Fri, 24 May 2019 23:04:10 GMT
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Pragma: no-cache
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Server: Microsoft-HTTPAPI/2.0
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: Strict-Transport-Security: max-age=31536000; includeSubDomains
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: X-Content-Type-Options: nosniff
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: X-Ms-Correlation-Request-Id: c104b849-2bdf-4009-b92f-a9f609901f32
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: X-Ms-Ratelimit-Remaining-Tenant-Reads: 11999
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: X-Ms-Request-Id: c104b849-2bdf-4009-b92f-a9f609901f32
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: X-Ms-Routing-Request-Id: CENTRALUS:20190524T230411Z:c104b849-2bdf-4009-b92f-a9f609901f32
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: panic: runtime error: index out of range
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: goroutine 121 [running]:
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandAzureRmCosmosDBAccountConsistencyPolicy(0xc000704e00, 0xc0008a839d)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_cosmosdb_account.go:777 +0x2e7
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmCosmosDbAccountCreate(0xc000704e00, 0x2ede3c0, 0xc00165e000, 0xc000704e00, 0x0)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_cosmosdb_account.go:362 +0x9d2
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0016ef080, 0xc000d56320, 0xc0009b3520, 0x2ede3c0, 0xc00165e000, 0x40b801, 0xc0010b4b80, 0x4c2cec)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:286 +0x363
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc001645a00, 0xc000d562d0, 0xc000d56320, 0xc0009b3520, 0xc0004f0000, 0x18, 0x7f01ef66d440)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:285 +0x9c
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc0010ad200, 0xc0009b34a0, 0xc00034e080, 0x0, 0x0)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:530 +0x57
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: reflect.Value.call(0xc000176900, 0xc000099c70, 0x13, 0x2fa522d, 0x4, 0xc0010b4f18, 0x3, 0x3, 0xc00019e880, 0xc000aa9410, ...)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/goenv/versions/1.11.6/src/reflect/value.go:447 +0x454
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: reflect.Value.Call(0xc000176900, 0xc000099c70, 0x13, 0xc0006e1f18, 0x3, 0x3, 0x1000200000000, 0xc000aa9400, 0x0)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/goenv/versions/1.11.6/src/reflect/value.go:308 +0xa4
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: net/rpc.(*service).call(0xc000aa8280, 0xc000826b40, 0xc0007fe310, 0xc0007fe320, 0xc001645a80, 0xc000680680, 0x2a84b40, 0xc0009b34a0, 0x16, 0x2a84b80, ...)
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/goenv/versions/1.11.6/src/net/rpc/server.go:384 +0x14e
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: created by net/rpc.(*Server).ServeCodec
[DEBUG] plugin.terraform-provider-azurerm_v1.28.0_x4: 	/opt/goenv/versions/1.11.6/src/net/rpc/server.go:481 +0x47e

### Panic Output

2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalWriteState
2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalApplyProvisioners
2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalIf
2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalWriteState
2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalWriteDiff
2019/05/24 23:04:11 [TRACE] root.cosmosdb: eval: *terraform.EvalApplyPost
2019/05/24 23:04:11 [ERROR] root.cosmosdb: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

azurerm_cosmosdb_account.db: unexpected EOF
2019-05-24T23:04:11.113Z [DEBUG] plugin: plugin process exited: path=/root/repo/spokane/infra/terraform/subscription/dev-engineering/dev-eus2-store-rg/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.28.0_x4
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalWriteState
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalApplyProvisioners
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalIf
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalWriteState
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalWriteDiff
2019/05/24 23:04:11 [TRACE] root.log_analytics: eval: *terraform.EvalApplyPost
2019/05/24 23:04:11 [ERROR] root.log_analytics: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

azurerm_log_analytics_workspace.la: unexpected EOF
2019/05/24 23:04:11 [ERROR] root.log_analytics: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

azurerm_log_analytics_workspace.la: unexpected EOF
2019/05/24 23:04:11 [TRACE] [walkApply] Exiting eval tree: module.log_analytics.azurerm_log_analytics_workspace.la
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.output.workspace_id"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.azurerm_log_analytics_solution.la[1]"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.azurerm_log_analytics_solution.la[0]"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.output.secondary_shared_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.output.primary_shared_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.log_analytics.output.portal_url"
2019/05/24 23:04:11 [ERROR] root.cosmosdb: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

azurerm_cosmosdb_account.db: unexpected EOF
2019/05/24 23:04:11 [TRACE] [walkApply] Exiting eval tree: module.cosmosdb.azurerm_cosmosdb_account.db
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.secondary_readonly_master_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.write_endpoints"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.primary_readonly_master_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.read_endpoints"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.primary_master_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.connection_strings"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.endpoint"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.name"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "module.cosmosdb.output.secondary_master_key"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/05/24 23:04:11 [TRACE] dag/walk: upstream errored, not walking "root"
2019/05/24 23:04:11 [TRACE] Preserving existing state lineage "c435d8ad-095d-8454-09c6-cdc37d4a5bf0"
2019/05/24 23:04:11 [TRACE] Preserving existing state lineage "c435d8ad-095d-8454-09c6-cdc37d4a5bf0"
2019/05/24 23:04:11 [TRACE] Preserving existing state lineage "c435d8ad-095d-8454-09c6-cdc37d4a5bf0"
2019/05/24 23:04:11 [TRACE] Preserving existing state lineage "c435d8ad-095d-8454-09c6-cdc37d4a5bf0"
2019/05/24 23:04:11 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-05-24T23:04:11.118Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"


### Expected Behavior

Created azure cosmosdb

### Actual Behavior


### Steps to Reproduce

1. `terraform init`
1. `terraform apply --auto-approve`

### Important Factoids

Public

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azurerm/resource_arm_cosmosdb_account.go
+++ b/azurerm/resource_arm_cosmosdb_account.go
@@ -774,7 +774,11 @@ func resourceArmCosmosDbAccountApiUpsert(client documentdb.DatabaseAccountsClien
 }
 
 func expandAzureRmCosmosDBAccountConsistencyPolicy(d *schema.ResourceData) *documentdb.ConsistencyPolicy {
-	input := d.Get("consistency_policy").([]interface{})[0].(map[string]interface{})
+	i := d.Get("consistency_policy").([]interface{})
+	if len(i) <= 0 {
+		return nil
+	}
+	input := i[0].(map[string]interface{})
 
 	consistencyLevel := input["consistency_level"].(string)
 	policy := documentdb.ConsistencyPolicy{

--- a/azurerm/resource_arm_cosmosdb_account.go
+++ b/azurerm/resource_arm_cosmosdb_account.go
@@ -775,7 +775,7 @@ func resourceArmCosmosDbAccountApiUpsert(client documentdb.DatabaseAccountsClien
 
 func expandAzureRmCosmosDBAccountConsistencyPolicy(d *schema.ResourceData) *documentdb.ConsistencyPolicy {
 	i := d.Get("consistency_policy").([]interface{})
-	if len(i) <= 0 {
+	if len(i) <= 0 || i[0] == nil {
 		return nil
 	}
 	input := i[0].(map[string]interface{})

